modules/videopress/utility-functions.php

```diff
@@ -637,13 +639,48 @@ function video_image_url_by_guid( $guid, $format ) {
  * Using a GUID, find a post.
  *
  * @param string $guid
- * @return WP_Post
+ * @return WP_Post|false The post for that guid, or false if none is found.
  */
 function video_get_post_by_guid( $guid ) {
+	$cache_key   = 'video_get_post_by_guid_' . $guid;
+	$cache_group = 'videopress';
+	$cached_post = wp_cache_get( $cache_key, $cache_group );
+
+	if ( is_object( $cached_post ) && 'WP_Post' === get_class( $cached_post ) ) {
+		return $cached_post;
+	}
+
+	$post_id = video_get_post_id_by_guid( $guid );
+
+	if ( is_int( $post_id ) ) {
+		$post = get_post( $post_id );
+		wp_cache_set( $cache_key, $post, $cache_group, HOUR_IN_SECONDS );
+
+		return $post;
+	}
+
+	return false;
+}
+
+/**
+ * Using a GUID, find the associated post ID.
+ *
+ * @param string $guid The guid to look for the post ID of.
+ * @return int|false The post ID for that guid, or false if none is found.
+ */
+function video_get_post_id_by_guid( $guid ) {
+	$cache_key = 'video_get_post_id_by_guid_' . $guid;
+	$cached_id = get_transient( $cache_key );
+
+	if ( is_int( $cached_id ) ) {
+		return $cached_id;
+	}
+
 	$args = array(
 		'post_type'      => 'attachment',
 		'post_mime_type' => 'video/videopress',
 		'post_status'    => 'inherit',
+		'no_found_rows'  => true,
 		'meta_query'     => array(
```

Comment:
As we are only using the ID here, we can use the ` 'fields' => 'ids' ` param in the query to SELECT only the ID from the database.

Reply:
Ah, sorry for missing that. It's added in https://github.com/Automattic/jetpack/commit/64918b1371d1dde532e7bd2e5f467d84ba3f32e4#diff-66c3e0655a6e9d684e7898c2c70171c3R685